feat: ascanrulesAlpha: WebCacheDeception use ComparableResponse

kingthorin opened the pull request:
- Update commonlib dependency version in build file.
- Add change note to CHANGELOG.md
- WebCacheDeceptionScanRule > Leverage the new component.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

kingthorin commented on addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/WebCacheDeceptionScanRule.java:
```diff
@@ -87,12 +85,15 @@ public void scan() {
+                    ComparableResponse unAuthCompResp =
+                            new ComparableResponse(unauthorisedCachedResponse, null);
+                    float similarity = authCompResp.compareWith(unAuthCompResp);
+                    if (similarity >= 0.75) {
```
This is a made up threshold. All of the tests seem to pan out to 1.0 or 0.0, and I didn't see an easy way to appropriately make partial matches. (I probably just didn't look hard enough :cry:)

kingthorin replied:
@bettercalln1ck any thoughts on either what a good match threshold would be (0 being totally different and 1.0 being very similar)? Or how to appropriately cause partial matches in the UnitTests?

kingthorin replied:
I've tweaked this to 80 after looking at some other things over the weekend.

kingthorin commented:
I believe this is ready for review.

thc202 commented on addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/WebCacheDeceptionScanRule.java:
```diff
@@ -87,12 +85,15 @@ public void scan() {
+                    ComparableResponse unAuthCompResp =
+                            new ComparableResponse(unauthorisedCachedResponse, null);
+                    float similarity = authCompResp.compareWith(unAuthCompResp);
+                    if (similarity >= 0.80) {
```
Maybe extract a constant or keep the `checkSimilarity` method.

thc202 commented:
I guess the timeout annotation in the test method could be removed, or the value reduced.

thc202 commented on addOns/ascanrulesAlpha/CHANGELOG.md:
```diff
@@ -4,7 +4,8 @@ All notable changes to this add-on will be documented in this file.
 ## Unreleased
-
+### Changed
+- The Web Cache Deception scan rule now uses a comparison mechanism which should be more performant.
```
Could mention zaproxy/zaproxy#6655.

kingthorin commented:
Extracted a constant for the similarity threshold, removed the intermediate store of the similarity value, reduced the timing unit test annotation to 1s, and added a check to skip assessing messages that weren't successful in the first place (a well as a related unittest).

thc202 commented:
Thank you!
